skbio/tree/tests/test_tree.py

```diff
@@ -633,6 +633,18 @@ def test_set_max_distance(self):
         self.assertEqual(tip_a[0] + tip_b[0], 1.6)
         self.assertEqual(sorted([tip_a[1].name, tip_b[1].name]), ['b', 'e'])
 
+    def test_set_max_distance_tie_bug(self):
```

Comment:
Yes, we need to have a test of the public function that was previously failing when provided with valid data, to protect against a similar bug coming back in the future if we change how this method works internally.